File: 700/713 - Subarray Product Less Than K.c
int numSubarrayProductLessThanK(int* nums, int numsSize, int k) {
    int ret = 0, prod = 1, first = 0;
    
    if (k > 1) {
        for (int i = 0; i < numsSize; ++i) {
            prod *= nums[i];

            while (prod >= k) {
                prod /= nums[first++];
            }

            ret += i - first + 1;
        }
    }
    
    return ret;
}

File: csharp/713.subarray-product-less-than-k.cs
/*
 * @lc app=leetcode id=713 lang=csharp
 *
 * [713] Subarray Product Less Than K
 */

// @lc code=start
public class Solution {
    public int NumSubarrayProductLessThanK(int[] nums, int k) {
        if (k <= 1) {
            return 0;
        }
        int ret = 0, prod = 1, i = 0, n = nums.Length;
        for (int j = 0; j < n; ++j) {
            prod *= nums[j];
            while (prod >= k) {
                prod /= nums[i];
                ++i;
            }
            ret += j - i + 1;
        }
        return ret;
    }
}
// @lc code=end

